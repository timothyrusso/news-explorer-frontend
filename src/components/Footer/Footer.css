.footer {
  width: 100%;
  min-height: 80px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.footer__copyright {
  font-family: 'Roboto', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.38;
  padding-left: 104px;
  margin: 0;
  color: #B6BCBF;
}

.footer__wrapper {
  padding-right: 104px;
  display: flex;
}

.footer__link {
  color: #1A1B22;
  font-family: 'Roboto', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  padding: 0 24px;
  transition: all linear 0.3s;
  text-decoration: none;
}

.footer__link:hover {
  cursor: pointer;
  opacity: 0.7;
}

.footer__icon {
  max-width: 24px;
  padding-left: 24px;
  transition: all linear 0.3s;
}

.footer__icon:hover {
  cursor: pointer;
  opacity: 0.7;
}

@media screen and (max-width: 910px) {
  .footer__copyright {
    padding-left: 40px;
  }

  .footer__wrapper {
    padding-right: 40px;
  }

  .footer__link {
    padding: 0 0 0 24px;
  }
}

@media screen and (max-width: 740px) {
  .footer {
    flex-direction: column;
    min-height: 176px;
  }

  .footer__wrapper {
    max-width: 400px;
    order: 1;
    margin: 20px 0 38px;
    width: 100%;
    padding: 0 16px;
    display: grid;
    grid-template-areas:
      'top icons'
      'bottom bottom';
    gap: 15px 80px;
    box-sizing: border-box;
  }

  .footer__copyright {
    order: 2;
    padding: 0 16px 22px;
  }

  .footer__link {
    grid-area: top;
    padding-left: 0;
  }

  .footer__link_type_mobile {
    grid-area: bottom;
  }

  .footer__icon-wrapper {
    display: flex;
    justify-content: flex-end;
    grid-area: icons;
  }

  .footer__icon {
    padding-left: 24px;
  }

}
