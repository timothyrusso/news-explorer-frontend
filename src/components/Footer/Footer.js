import "./Footer.css";
import linkedinIcon from "../../images/linkedin_icon.png";
import githubIcon from "../../images/github.svg";
import { Link } from "react-router-dom";

const Footer = () => {
  const date = new Date();
  const year = date.getFullYear();

  return (
    <footer className="footer">
      <p className="footer__copyright">
        Â© {year} NewsExplorer - Powered by News API
      </p>
      <div className="footer__wrapper">
        <Link to={"/"} className="footer__link">
          Home
        </Link>
        <a
          href="https://timothyrusso.tech/"
          target="_blank"
          rel="noreferrer"
          className="footer__link footer__link_type_mobile"
        >
          Portfolio
        </a>
        <div className="footer__icon-wrapper">
          <a
            href="https://github.com/timothyrusso"
            target="_blank"
            rel="noreferrer"
          >
            <img
              src={githubIcon}
              className="footer__icon"
              alt="GitHub logo"
            ></img>
          </a>
          <a
            href="https://www.linkedin.com/in/russotimothysoftwareengineer/"
            target="_blank"
            rel="noreferrer"
          >
            <img
              src={linkedinIcon}
              className="footer__icon"
              alt="LinkedIn Logo"
            ></img>
          </a>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
