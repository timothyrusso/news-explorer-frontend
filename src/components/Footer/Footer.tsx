import React from 'react';
import linkedinIcon from '../../images/linkedin_icon.png';
import githubIcon from '../../images/github.svg';
import { Link } from 'react-router-dom';
import './Footer.css';

const Footer = () => {
  const date = new Date();
  const year = date.getFullYear();

  return (
    <footer className="footer">
      <p className="footer__copyright">
        Â© {year} NewsExplorer - Powered by{' '}
        <a href="https://newsapi.org/" target="_blank" rel="noreferrer">
          News API
        </a>
      </p>
      <nav className="footer__wrapper">
        <Link to={'/'} className="footer__link">
          Home
        </Link>
        <a
          href="https://timothyrusso.dev/"
          target="_blank"
          rel="noreferrer"
          className="footer__link footer__link_type_mobile"
        >
          Portfolio
        </a>
        <ul className="footer__icon-wrapper">
          <a
            href="https://github.com/timothyrusso"
            target="_blank"
            rel="noreferrer"
          >
            <img
              src={githubIcon}
              className="footer__icon"
              alt="GitHub logo"
            ></img>
          </a>
          <a
            href="https://www.linkedin.com/in/russotimothysoftwareengineer/"
            target="_blank"
            rel="noreferrer"
          >
            <img
              src={linkedinIcon}
              className="footer__icon"
              alt="LinkedIn Logo"
            ></img>
          </a>
        </ul>
      </nav>
    </footer>
  );
};

export default Footer;
