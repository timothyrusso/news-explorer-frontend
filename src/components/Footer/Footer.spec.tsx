import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Footer from './Footer';

describe('Footer component', () => {
  it('Should render the Footer component', () => {
    const { asFragment } = render(
      <Router>
        <Footer />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
