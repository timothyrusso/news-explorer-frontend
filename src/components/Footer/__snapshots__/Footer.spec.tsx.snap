// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should render the Footer component 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <p
      class="footer__copyright"
    >
      Â© 2023 NewsExplorer - Powered by 
      <a
        href="https://newsapi.org/"
        rel="noreferrer"
        target="_blank"
      >
        News API
      </a>
    </p>
    <nav
      class="footer__wrapper"
    >
      <a
        class="footer__link"
        href="/"
      >
        Home
      </a>
      <a
        class="footer__link footer__link_type_mobile"
        href="https://timothyrusso.dev/"
        rel="noreferrer"
        target="_blank"
      >
        Portfolio
      </a>
      <ul
        class="footer__icon-wrapper"
      >
        <a
          href="https://github.com/timothyrusso"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub logo"
            class="footer__icon"
            src="test-file-stub"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/russotimothysoftwareengineer/"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="LinkedIn Logo"
            class="footer__icon"
            src="test-file-stub"
          />
        </a>
      </ul>
    </nav>
  </footer>
</DocumentFragment>
`;
